bricks/visual-builder/src/workbench-tree/index.tsx

```diff
@@ -0,0 +1,189 @@
+import React from "react";
+import { pick } from "lodash";
+import { EventEmitter, createDecorators } from "@next-core/element";
+import { ReactNextElement } from "@next-core/react-element";
+import type {
+  ActionClickDetail,
+  WorkbenchNodeData,
+  WorkbenchTreeAction,
+} from "../interfaces.js";
+import { WorkbenchActionsContext } from "../shared/workbench/WorkbenchActionsContext.js";
+import {
+  WorkbenchTree,
+  dropEmitProps,
+} from "../shared/workbench/WorkbenchTree.js";
+import { WorkbenchTreeContext } from "../shared/workbench/WorkbenchTreeContext.js";
+import { deepMatch } from "../utils/deepMatch.js";
+import treeStyleText from "./WorkbenchTree.shadow.css";
+import actionStyleText from "../shared/workbench-mini-action-bar/WorkbenchMiniActionBar.shadow.css";
+import "@next-core/theme";
+
+function defaultGetCollapsedId(node: WorkbenchNodeData): string | number {
+  return node.key;
+}
+
+const { defineElement, property, event } = createDecorators();
+
+/**
+ * @id next-builder.workbench-tree
+ * @author steve
+ * @history
+ * 1.x.0: 新增构件 `next-builder.workbench-tree`
+ * @docKind brick
+ * @noInheritDoc
+ */
+@defineElement("visual-builder.workbench-tree", {
+  styleTexts: [treeStyleText, actionStyleText],
+})
+class WorkbenchTreeElement extends ReactNextElement {
+  @property({ attribute: false })
+  accessor nodes: WorkbenchNodeData[];
+
+  @property({ attribute: false })
+  accessor actions: WorkbenchTreeAction[];
+
+  @property({ type: Boolean })
+  accessor actionsHidden: boolean;
+
+  @property()
+  accessor placeholder: string;
+
+  @property({ type: Boolean })
+  accessor isTransformName: boolean;
+
+  @property()
+  accessor searchPlaceholder: string;
+
+  @property({ type: Boolean })
+  accessor noSearch: boolean;
+
+  @property({ attribute: false })
+  accessor activeKey: string | number;
+
+  @property({ type: Boolean })
+  accessor showMatchedNodeOnly: boolean;
+
+  @property({ attribute: false })
+  accessor matchNodeDataFields: string | string[];
+
+  @property({ attribute: false })
+  accessor fixedActionsFor: Record<string, unknown> | Record<string, unknown>[];
+
+  @property({ type: Boolean })
+  accessor collapsible: boolean;
+
+  @property({ attribute: false })
+  accessor collapsedNodes: string[];
+
+  @property({ type: Boolean })
+  accessor allowDrag: boolean;
+
+  @property({ type: Boolean })
+  accessor allowDragToRoot: boolean;
+
+  @property({ type: Boolean })
+  accessor allowDragToInside: boolean;
+
+  @property({ type: String })
+  accessor nodeKey: string;
+
+  @property({ type: Boolean })
+  accessor skipNotify: boolean;
+
+  @event({ type: "action.click" })
+  accessor #actionClickEvent!: EventEmitter<ActionClickDetail>;
+
+  private _handleActionClick = (detail: ActionClickDetail): void => {
```

Comment:
v3 构件类建议使用 JS 原生私有字段 `#` 代替 TS `private`

```suggestion
  #handleActionClick = (detail: ActionClickDetail): void => {
```